test/utils/dirft1d.hpp

```diff
@@ -1,82 +1,124 @@
-#include "finufft/finufft_core.h"
-#include <iostream>
+#pragma once
 
-// This is basically a port of dirft1d.f from CMCL package, except with
-// the 1/nj prefactors for type-1 removed.
+#include <cmath>
+#include <complex>
+#ifdef __CUDACC__
+#include <thrust/complex.h>
+#endif
+
+// ------------------------------------------------------------
+// 1D type‑1 NUFFT, direct:
+//   f[m] = Σ_{j=0..nj-1} c[j] exp(i*iflag * k_m * x[j])
+//   k_m = -⌊ms/2⌋ + m
+// ------------------------------------------------------------
+template<typename BIGINT,
```

Comment:
I'm still unsure why the level of code rewrite was needed here. Wasn't there a simpler way to just declare FLT and BIGINT to avoid the finufft_core header dependency and allow GPU usage?